### ISSUE TITLE
Details component has unnecessary aria-label

### ISSUE THREAD

**Issue by rsese** [primer/view_components#3533]
👋 !  Coming here via this internal issue:

https://github.com/github/accessibility-audits/issues/11204

At a11y office hours (x-ref: https://github.com/github/accessibility/issues/8691#issuecomment-2939525246) it was suggested that we remove the aria-label on a details `<summary>` element because it's redundant:

>  * screen reader already communicates the expanded/collapsed state, so the label doesn't need to provide that information.
> * At the moment, the aria-label is overriding the visible label, and might ultimately conflict.
> * Correct solution is to remove the aria-label entirely.

I thought in our code we might have used `<details>` directly but found we use the Details component.  In my digging I see there's no way for a user to remove the `aria-label` because if none is provided, it falls back to default labels:

https://github.com/primer/view_components/blob/c3e0e7903253494428c5f047b4a409cd50b13fdc/app/components/primer/beta/details.rb#L35-L36

Dropped into the a11y slack channel about this and it was suggested to open an issue here about this problem: https://github.slack.com/archives/C0FSWLQ0Y/p1749573777492109
